### ISSUE TITLE
Dagit can't handle nan float values in configuration

### ISSUE THREAD

**Issue by d-rams** [dagster-io/dagster#5962]
## Summary
Configuration fields for ops should be able to take POD values such as `float`, `str`, `int` etc.  However, `float('Nan')` is not actually properly handled by Dagit.  In particular, an op with a default config field of nan will run properly if executed programmatically, but the Dagit UI will convert this value to the string `.nan` and then choke on this value which is no longer a float type. 

## Reproduction
A simple `example.py`:
```
from dagster import (
    job,
    op,
    AssetMaterialization,
    DagsterInstance,
    Output,
)


@op(name="is_config_na", config_schema={"val": float})
def is_config_na(context) -> bool:
    val = context.solid_config["val"]
    isna = val != val
    yield AssetMaterialization(asset_key="isna", description=str(isna))
    yield Output(isna)


default_config = {
    "solids": {"is_config_na": {"config": {"val": float("NaN")}}},
}

@job(config=default_config)
def isna_job():
    is_config_na()


if __name__ == "__main__":
    isna_job.execute_in_process(run_config=default_config, instance=DagsterInstance.local_temp())
```

This works fine executing via `python example.py`.  However, upon `dagit -f example.py` and looking to launch a run we see the following error:

## Dagit UI/UX Issue Screenshots
<img width="1670" alt="Screen Shot 2021-12-14 at 9 35 56 AM" src="https://user-images.githubusercontent.com/3892928/146051252-83fd555c-3b07-4470-98ce-2bd3bd9fc20f.png">

## Additional Info about Your Environment
tl;dr: dagster/dagit 0.13.9 w/ python 3.7.9
-------------------
# packages in anaconda environment:
```

#
# Name                    Version                   Build  Channel
alembic                   1.6.5                    pypi_0    pypi
aniso8601                 7.0.0                    pypi_0    pypi
appnope                   0.1.2                    pypi_0    pypi
argcomplete               1.12.3                   pypi_0    pypi
argon2-cffi               21.2.0                   pypi_0    pypi
argon2-cffi-bindings      21.2.0                   pypi_0    pypi
astroid                   2.5                      pypi_0    pypi
attrs                     21.2.0                   pypi_0    pypi
backcall                  0.2.0                    pypi_0    pypi
black                     21.9b0                   pypi_0    pypi
blas                      1.0                         mkl
bleach                    4.1.0                    pypi_0    pypi
blosc                     1.21.0               h2842e9f_0
bzip2                     1.0.8                h1de35cc_0
ca-certificates           2021.10.26           hecd8cb5_2
certifi                   2021.10.8        py37hecd8cb5_0
cffi                      1.15.0                   pypi_0    pypi
charset-normalizer        2.0.9                    pypi_0    pypi
click                     7.1.2                    pypi_0    pypi
coloredlogs               14.0                     pypi_0    pypi
coverage                  6.2                      pypi_0    pypi
croniter                  1.1.0                    pypi_0    pypi
cytoolz                   0.10.1           py37h9ed2024_0
dagit                     0.13.9                   pypi_0    pypi
dagster                   0.13.9                   pypi_0    pypi
dagster-graphql           0.13.9                   pypi_0    pypi
debugpy                   1.5.1                    pypi_0    pypi
decorator                 5.1.0                    pypi_0    pypi
defusedxml                0.7.1                    pypi_0    pypi
docstring-parser          0.13                     pypi_0    pypi
dunamai                   1.3.0                    pypi_0    pypi
entrypoints               0.3                      pypi_0    pypi
flask                     1.1.4                    pypi_0    pypi
flask-cors                3.0.10                   pypi_0    pypi
flask-graphql             2.0.1                    pypi_0    pypi
flask-sockets             0.2.1                    pypi_0    pypi
gevent                    21.8.0                   pypi_0    pypi
gevent-websocket          0.10.1                   pypi_0    pypi
gql                       2.0.0                    pypi_0    pypi
graphene                  2.1.9                    pypi_0    pypi
graphql-core              2.3.2                    pypi_0    pypi
graphql-relay             2.0.1                    pypi_0    pypi
graphql-server-core       1.2.0                    pypi_0    pypi
graphql-ws                0.3.1                    pypi_0    pypi
greenlet                  1.1.2                    pypi_0    pypi
grpcio                    1.42.0                   pypi_0    pypi
grpcio-health-checking    1.42.0                   pypi_0    pypi
hdf5                      1.10.4               hfa1e0ec_0
humanfriendly             10.0                     pypi_0    pypi
hypothesis                5.30.0                   pypi_0    pypi
idna                      3.3                      pypi_0    pypi
importlab                 0.6.1                    pypi_0    pypi
importlib-metadata        1.7.0                    pypi_0    pypi
importlib-resources       5.4.0                    pypi_0    pypi
iniconfig                 1.1.1                    pypi_0    pypi
intel-openmp              2021.4.0          hecd8cb5_3538
ipykernel                 6.6.0                    pypi_0    pypi
ipython                   7.30.1                   pypi_0    pypi
ipython-genutils          0.2.0                    pypi_0    pypi
ipywidgets                7.6.5                    pypi_0    pypi
isort                     5.10.1                   pypi_0    pypi
itsdangerous              1.1.0                    pypi_0    pypi
jedi                      0.18.1                   pypi_0    pypi
jinja2                    2.11.3                   pypi_0    pypi
jsonschema                4.2.1                    pypi_0    pypi
junit-xml                 1.9                      pypi_0    pypi
jupyter                   1.0.0                    pypi_0    pypi
jupyter-client            7.1.0                    pypi_0    pypi
jupyter-console           6.4.0                    pypi_0    pypi
jupyter-core              4.9.1                    pypi_0    pypi
jupyterlab-widgets        1.0.2                    pypi_0    pypi
lazy-object-proxy         1.6.0                    pypi_0    pypi
libcxx                    12.0.0               h2f01273_0
libffi                    3.3                  hb1e8313_2
libgfortran               3.0.1                h93005f0_2
lz4-c                     1.9.3                h23ab428_1
lzo                       2.10                 haf1e3a3_2
mako                      1.1.6                    pypi_0    pypi
markupsafe                2.0.1                    pypi_0    pypi
matplotlib-inline         0.1.3                    pypi_0    pypi
mccabe                    0.6.1                    pypi_0    pypi
mistune                   0.8.4                    pypi_0    pypi
mkl                       2019.4                      233
mkl-service               2.3.0            py37h9ed2024_0
mkl_fft                   1.3.0            py37ha059aab_0
mkl_random                1.1.1            py37h959d312_0
mock                      4.0.3              pyhd3eb1b0_0
mypy-extensions           0.4.3                    pypi_0    pypi
nbconvert                 5.6.1                    pypi_0    pypi
nbformat                  5.1.3                    pypi_0    pypi
ncurses                   6.3                  hca72f7f_2
nest-asyncio              1.5.4                    pypi_0    pypi
networkx                  2.6.3                    pypi_0    pypi
ninja                     1.10.2.3                 pypi_0    pypi
notebook                  6.4.6                    pypi_0    pypi
numexpr                   2.7.3            py37h16bde0e_0
numpy                     1.19.1           py37h3b9f5b6_0
numpy-base                1.19.1           py37hcfb5961_0
openssl                   1.1.1l               h9ed2024_0
packaging                 21.3                     pypi_0    pypi
pandas                    1.2.4            py37hb2f4e1b_0
pandera                   0.7.2                    pypi_0    pypi
pandocfilters             1.5.0                    pypi_0    pypi
parso                     0.8.3                    pypi_0    pypi
pathspec                  0.9.0                    pypi_0    pypi
pendulum                  2.1.2                    pypi_0    pypi
pexpect                   4.8.0                    pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pip                       20.2.2           py37hecd8cb5_0
platformdirs              2.4.0                    pypi_0    pypi
plombier                  0.0.0.post363.dev0+a75c478.dirty           dev_0    <develop>
pluggy                    0.13.1                   pypi_0    pypi
prometheus-client         0.12.0                   pypi_0    pypi
promise                   2.3                      pypi_0    pypi
prompt-toolkit            3.0.24                   pypi_0    pypi
protobuf                  3.19.1                   pypi_0    pypi
ptyprocess                0.7.0                    pypi_0    pypi
py                        1.11.0                   pypi_0    pypi
pyarrow                   6.0.1                    pypi_0    pypi
pycparser                 2.21                     pypi_0    pypi
pydantic                  1.6.1                    pypi_0    pypi
pygments                  2.10.0                   pypi_0    pypi
pylint                    2.6.0                    pypi_0    pypi
pylint-junit              0.2.0                    pypi_0    pypi
pyparsing                 3.0.6                    pypi_0    pypi
pyrsistent                0.18.0                   pypi_0    pypi
pytables                  3.6.1            py37h5bccee9_0
pytest                    6.1.0                    pypi_0    pypi
pytest-cov                2.10.1                   pypi_0    pypi
python                    3.7.9                h26836e1_0
python-dateutil           2.8.2              pyhd3eb1b0_0
python-editor             1.0.4                    pypi_0    pypi
pytype                    2020.9.29                pypi_0    pypi
pytz                      2021.3             pyhd3eb1b0_0
pytzdata                  2020.1                   pypi_0    pypi
pyyaml                    6.0                      pypi_0    pypi
pyzmq                     22.3.0                   pypi_0    pypi
qtconsole                 5.2.1                    pypi_0    pypi
qtpy                      1.11.3                   pypi_0    pypi
readline                  8.1                  h9ed2024_0
regex                     2021.11.10               pypi_0    pypi
requests                  2.26.0                   pypi_0    pypi
rx                        1.6.1                    pypi_0    pypi
scipy                     1.5.2            py37h912ce22_0
send2trash                1.8.0                    pypi_0    pypi
setuptools                49.6.0                   py37_1
six                       1.16.0             pyhd3eb1b0_0
sortedcontainers          2.4.0                    pypi_0    pypi
sqlalchemy                1.4.28                   pypi_0    pypi
sqlite                    3.36.0               hce871da_0
tabulate                  0.8.9                    pypi_0    pypi
terminado                 0.12.1                   pypi_0    pypi
testpath                  0.5.0                    pypi_0    pypi
tk                        8.6.11               h7bc2e8c_0
toml                      0.10.2                   pypi_0    pypi
tomli                     1.2.2                    pypi_0    pypi
toolz                     0.11.2             pyhd3eb1b0_0
toposort                  1.7                      pypi_0    pypi
tornado                   6.1                      pypi_0    pypi
tqdm                      4.62.3                   pypi_0    pypi
traitlets                 5.1.1                    pypi_0    pypi
typed-ast                 1.5.1                    pypi_0    pypi
typing-compat             0.1.0                    pypi_0    pypi
typing-extensions         4.0.1                    pypi_0    pypi
typing-inspect            0.7.1                    pypi_0    pypi
urllib3                   1.26.7                   pypi_0    pypi
watchdog                  2.1.6                    pypi_0    pypi
wcwidth                   0.2.5                    pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
werkzeug                  1.0.1                    pypi_0    pypi
wheel                     0.37.0             pyhd3eb1b0_1
widgetsnbextension        3.5.2                    pypi_0    pypi
wrapt                     1.13.3                   pypi_0    pypi
xz                        5.2.5                h1de35cc_0
zipp                      3.6.0                    pypi_0    pypi
zlib                      1.2.11               h1de35cc_3
zope-event                4.5.0                    pypi_0    pypi
zope-interface            5.4.0                    pypi_0    pypi
zstd                      1.4.9                h322a384_0
```

---
#### Message from the maintainers:
Impacted by this bug? Give it a 👍. We factor engagement into prioritization.

**Comment by hellendag** [comment]
Thanks for the code snippet, super helpful.

**Comment by hellendag** [comment]
The gist is that we are taking yaml run config, turning it into JS for CodeMirror linting in the config editor, then sending that JS back as a GraphQL argument in the validation step. Unfortunately, since `NaN` is not valid in JSON, when the argument is serialized for the request, `JSON.stringify` replaces the `NaN` with a `null`, and the information is lost.

We could instead send the yaml string as the GraphQL argument, and thus not have to worry about JSON serialization. This will require some changes on the Python side, but it seems like a worthwhile change. @smackesey is taking a look.

**Comment by cmpadden** [comment]
Closing as this has been resolved.
